pkg/core/resources/apis/meshservice/api/v1alpha1/meshservice.go

```diff
@@ -53,8 +63,70 @@ type TLS struct {
 	Status TLSStatus `json:"status,omitempty"`
 }
 
+type HostnameGeneratorRef struct {
+	CoreName string `json:"name"`
+}
+
+const (
+	GeneratedCondition string = "Generated"
+)
+
+const (
+	GeneratedReason     string = "Generated"
+	TemplateErrorReason string = "TemplateError"
+)
+
+type Condition struct {
```

Comment:
Should this be a common object? I think `MeshExternalService` could have the same structure

Reply:
I'll leave that to the PR that adds MeshExternalService